### PR TITLE
fix: styled-components no longer  overridden after page reload

### PR DESCRIPTION
**Pull request by TechSupportz**
Fixed a bug which resulted in styled components to stop working when the browser was refreshed

Changes:
- Added `styledComponents: true` under compiler options in `next.config.js`
- Replaced `_document.tsx` with [this example](https://github.com/vercel/next.js/blob/canary/examples/with-styled-components/pages/_document.tsx) which is recommend by the [styled-components documentation](https://styled-components.com/docs/advanced#nextjs)

### CODE REVIEW

File: src/styles/globals.css [hunk 1]
```diff
@@ -1,24 +1,3 @@
...
-    box-sizing: border-box;
-    padding: 0;
-    margin: 0;
-}
 
-html,
-body {
-    max-width: 100vw;
-    overflow-x: hidden;
-}
```
Comment (Shockch4rge) [inline src/styles/globals.css:13]: Could probably remove this

File: src/styles/globals.css [hunk 2]
```diff
@@ -1,24 +1,3 @@
...
-a {
-    color: inherit;
-    text-decoration: none;
-}
-
-@media (prefers-color-scheme: dark) {
-    html {
-        color-scheme: dark;
-    }
-}
```
Comment (Shockch4rge) [inline src/styles/globals.css:24]: This config is required for CSS reset.

**Review by Shockch4rge — CHANGES_REQUESTED** [review]

File: src/styles/globals.css (hunk 1 above)
Comment (Shockch4rge) [inline src/styles/globals.css:13]: This can be removed in the meantime.

File: src/styles/globals.css (hunk 2 above)
Comment (shockch4rge) [inline src/styles/globals.css]: We'll need to keep this for a CSS reset.

File: src/styles/globals.css (hunk 1 above)
Comment (shockch4rge) [inline src/styles/globals.css:13]: This can be removed in the meantime.

**Review by shockch4rge — CHANGES_REQUESTED** [review]

**Review by shockch4rge — APPROVED** [review]
Looks good

**Pull request merged**